### ISSUE TITLE
Populating an entity with eager enabled causes 'Maximum call stack size exceeded' exporting with toJSON()

### ISSUE THREAD

**Issue by jsprw** [mikro-orm/mikro-orm#205]
**Describe the bug**
When finding entities with populate: true or populate: ["entity"] causes a 'Maximum call stack size exceeded' error.

Bonus: how do you initialize @OneToOne relations from the object if it isn't loaded during the query, it isn't like a Collection?

```
//Entity1

@OneToOne(() => Entity2, e => e.joinprop, { eager: true })
    ent2: Entity2;

//Entity3
@ManyToOne(() => Entity1)
    ent1: Entity1;

// service

async list(
    ent1: Entity1,
    limit: number,
    offset: number
): Promise<[Entity3[], number]> {
    const connection = await Database.getConnection();
    return connection.em
        .getRepository(Entity3)
        .findAndCount({ ent1 }, { limit, offset, populate: ["ent1"] }); //or populate:true
}
```

**Stack trace**
```
Unhandled Promise rejection: Maximum call stack size exceeded ; Zone: <root> ; Task: Promise.then ; Value: RangeError: Maximum call stack size exceeded
    at Array Iterator.next (<anonymous>)
    at Function.processEntity (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:51:56)
    at Function.processProperty (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:43:38)
    at C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:21:51
    at Array.map (<anonymous>)
    at Function.toObject (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:21:14)
    at Entity1.prototype.toJSON (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityHelper.js:28:62)
    at Function.processEntity (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:52:26)
    at Function.processProperty (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:43:38)
    at C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:21:51
    at Array.map (<anonymous>)
    at Function.toObject (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:21:14)
    at Entity2.prototype.toJSON (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityHelper.js:28:62)
    at Function.processEntity (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:52:26)
    at Function.processProperty (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:43:38)
    at C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:21:51 RangeError: Maximum call stack size exceeded
    at Array Iterator.next (<anonymous>)
    at Function.processEntity (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:51:56)
    at Function.processProperty (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:43:38)
    at C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:21:51
    at Array.map (<anonymous>)
    at Function.toObject (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:21:14)
    at Entity1.prototype.toJSON (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityHelper.js:28:62)
    at Function.processEntity (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:52:26)
    at Function.processProperty (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:43:38)
    at C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:21:51
    at Array.map (<anonymous>)
    at Function.toObject (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:21:14)
    at Entity2.prototype.toJSON (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityHelper.js:28:62)
    at Function.processEntity (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:52:26)
    at Function.processProperty (C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:43:38)
    at C:\Users\XX\Repositories\XXProj\node_modules\mikro-orm\dist\entity\EntityTransformer.js:21:51
```

**To Reproduce**
1. Have an entity that has eager enabled
2. Find with populate enabled
3. Crash.

**Expected behavior**
Return an object with all entities initialized.

**Versions**

| Dependency | Version |
| - | - |
| node | 12.9.1 |
| typescript | 3.5.3 |
| mikro-orm | 3.0.0-alpha.27 |
| your-driver | mysql2 |

**Comment by jsprw** [comment]
> This is caused by serializing the entity, so as a workaround you could handle the recursion check manually in custom toJSON() implementation.

Yes. I see now, has to do with the object exporting; changed the title. Will have a proper look at this tomorrow, thank you. 

> Not sure I understand - are you asking how to do something or how something works in MikroORM internally? :] There are no collections in 1:1 relations, just one (owning side) pointing to the inverse via unique foreign key.

I was wondering if Mikro-ORM offered such feature that you can initialize OneToOne as it just returns a property with the number 1 if not pop. But... I based it on the JSON output instead of the objects in Node.. I have not asked anything :)

**Comment by B4nan** [comment]
> I was wondering if Mikro-ORM offered such feature that you can initialize OneToOne as it just returns a property with the number 1 if not pop. But... I based it on the JSON output instead of the objects in Node.. I have not asked anything :)

Maybe you mean this?

```typescript
const e1 = em.findOne(Ent1, 1, { populate: true });
wrap(e1.foo).populated(false); // foo is 1:1 property that got populated and would cause the recursion error
```

That way you marked the reference as 'not populated by user', meaning it will be serialized as if it were not populated at all.

**Comment by B4nan** [comment]
Fixed in `3.0.0-alpha.28`.
